### DISCUSSION TITLE
How to do housekeeping on PV based docker cache

### DISCUSSION THREAD

**Discussion by yves-harms** [actions/actions-runner-controller#1896] [Questions]
As described in https://github.com/actions-runner-controller/actions-runner-controller/blob/master/docs/detailed-docs.md#docker-image-layers-caching the advice is to use dynamically requested PVs to cache docker images.

How do you guys do housekeeping on these volumes to avoid them growing too much in size over time?

**Comment by Diontsoumas** [comment]
I am also wondering about that. Currently, we just delete the PVs once in a while (we have some alerts on their size), but it's not working very nicely.

We have also experimented with different CSI solutions on GCP, as having around a big pool of PVs can be quite expensive if you scale your jobs, but not luck on that front either (check https://github.com/actions-runner-controller/actions-runner-controller/discussions/1795)
